daemon/cmd/datapath.go

```diff
@@ -458,6 +465,31 @@ func setupIPSec() (int, uint8, error) {
 	return authKeySize, spi, nil
 }
 
+func setupIPCacheVTEPMapping() error {
+	var (
+		worldID    uint32 = 2 // VTEP use world id "2"
+		encryptKey uint8  = 0 // no encrypt support
+	)
+
+	for i := 0; i < len(option.Config.VtepEndpoints); i++ {
+		log.WithFields(logrus.Fields{
+			logfields.IPAddr: option.Config.VtepEndpoints[i],
+			"allocCIDR":      option.Config.VtepCIDRs[i],
+		}).Info("Updating ipcache map entry")
+
+		err := ipcachemap.UpdateIPCacheMapping(option.Config.VtepCIDRs[i], option.Config.VtepEndpoints[i], worldID, option.Config.VtepMACs[i], encryptKey)
+		if err != nil {
+			log.WithError(err).WithFields(logrus.Fields{
+				"allocCIDR": option.Config.VtepCIDRs[i],
+			}).Error("bpf: Unable to update in IPCache map")
+			return err
```

Comment:
The caller will already handle the error, so there's no need to log _and_ return the error. I suggest wrapping the error here instead to provide the context about where this is coming from:

```suggestion
			return fmt.Errorf("Unable to set up VTEP mappings: %w", err)
```

Reply:
done